src/NuGet.Core/NuGet.CommandLine.XPlat/PushCommand.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Dnx.Runtime.Common.CommandLine;
+using NuGet.Commands;
+using NuGet.Configuration;
+using NuGet.Logging;
+using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Core.v3;
+
+namespace NuGet.CommandLine.XPlat
+{
+    class PushCommand:Command
+    {
+        public PushCommand(CommandLineApplication app, Func<ILogger> getLogger)
+        {
+            app.Command("push", push =>
+            {
+                push.Description = "Push to remote source";
+
+                var source = push.Option(
+                    "-s|--source <source>",
+                    Strings.Source_Description,
+                    CommandOptionType.SingleValue);
+
+                var timeout = push.Option(
+                    "-t|--timeout <timeout>",
+                    Strings.Push_Timeout_Description,
+                    CommandOptionType.SingleValue);
+
+                var argRoot = push.Argument(
+                    "[root]",
+                    Strings.Push_Package_ApiKey_Description,
+                    multipleValues: true);
+
+                push.OnExecute(async () =>
+                {
+                    ILogger logger = getLogger();
+
+                    ISettings setting = Settings.LoadDefaultSettings(Path.GetFullPath("."),
+                                configFileName: null,
+                                machineWideSettings: null);
+                    PackageUpdateResource pushCommandResource = await GetPushCommandResource(source, setting);
+
+                    string packagePath = string.Empty;
+                    if (argRoot.Values.Count > 0)
+                    {
+                        packagePath = argRoot.Values[0];
+                    }
+
+                    string apiKey = string.Empty;
+                    if (argRoot.Values.Count > 1)
```

Comment:
There's really no expectation for xplat to take identical command line parameters as nuget.exe. They should be similar, but not interchangeable.